### DISCUSSION TITLE
Add a `current block` variable in task

### DISCUSSION THREAD

**Discussion by failable** [zed-industries/zed#24681] [Feature Requests]
### Summary

Add a `current block` variable in task. Blocks are separated by newlines (or tree-sitter query expansion)

**Comment by notpeter** [comment]
Can you provide an example of how you would like this to work in your language of choice?  E.g. how would the "current block" be defined -- e.g. the current fold?

**Comment by failable** [comment]
@notpeter 

> e.g. the current fold?

> Blocks are separated by newlines.

No fancy thing. By blocks, just simply some 'paragraph's separated by newlines. It does not matter what current language is. e.g.

```
# This is some text                    <-- block1
# here

class A:                                       <--block2
    def __init__(self):
        self.x = 1

def fun():             < --block3
    pass

def fun2():    <-- block4
    a = 1

    b = 2        <-- block5, though don't make sense in the syntax view
```

Or 

``` 
[build-system]                                         <-- block1
requires = ["hatchling"]
build-backend = "hatchling.build"

[tool.rye]                                               <-- block2
virutal = true
managed = true
dev-dependencies = []
```

Or 

```
# saferx            <-- block1

Describe your project here.          <-- block2
```

**Comment by notpeter** [comment]
Gotcha. Newlines would be pretty specific to a subset of languages (python, toml, ini, etc). The reason I mentioned folds is because we potentially already have this context from tree-sitter so it might more sense to leverage that for a more language-agnostic capability.

<img width="317" alt="Screenshot 2024-08-15 at 11 44 32" src="https://github.com/user-attachments/assets/60be552b-71f3-4cbd-989b-b129ee9f5dc9">

Anyways, thanks for reporting.

**Comment by failable** [comment]
😅My 'X-problem' is I just simply want to get the current block content to run some external tasks like making HTTP request in a `.http` file without selection.

```http
# Get sources
POST http://localhost:7777/v1/messages/sources

# Get groups
POST http://localhost:7777/v1/messages/groups
{
   "source": "telegram"
}

# User Documents
POST http://localhost:9999/v1/user_documents
Content-Type: application/json
{
    "user_id": "66a89a81dee17b00119020b8"
}
```

Or call an external task to translate some paragraph, although we currently do not have the ability to insert the task output or replace with the task output (https://github.com/zed-industries/zed/issues/12598, https://github.com/zed-industries/zed/issues/9445).

BTW, I also considered requesting `ZED_*` variables where * could be class, function, etc but I wonder if there are too much variables for different syntax nodes and haven't do that. (We already have `ZED_SYMBOL`.)
